lib/matplotlib/text.py

```diff
@@ -1380,6 +1391,30 @@ def set_fontname(self, fontname):
         """
         self.set_fontfamily(fontname)
 
+    def _ha_for_angle(self, angle, is_tick_top_enabled):
+        """
+        Determines horizontal alignment ('ha') based on the angle of rotation
+        in degrees. Adjusts for is_tick_top_enabled.
+        """
+        if (angle < 5 or 85 <= angle < 105 or 355 <= angle < 360 or
+                170 <= angle < 190 or 265 <= angle < 275):
+            return 'center'
+        elif 5 <= angle < 85 or 190 <= angle < 265:
+            return 'left' if is_tick_top_enabled else 'right'
+        return 'right' if is_tick_top_enabled else 'left'
```

Comment:
```suggestion
    def _ha_for_angle(self, angle):
        """
        Determines horizontal alignment ('ha') for rotation_mode "xtick" based on
        the angle of rotation in degrees and the vertical alignment.
        """
        anchor_at_bottom = self.get_verticalalignment() == 'bottom'
        if (angle < 5 or 85 <= angle < 105 or 355 <= angle < 360 or
                170 <= angle < 190 or 265 <= angle < 275):
            return 'center'
        elif 5 <= angle < 85 or 190 <= angle < 265:
            return 'left' if anchor_at_bottom else 'right'
        return 'right' if anchor_at_bottom else 'left'
```

If one was really pedantic, this xtick rotation style is only defined for verticalalignment 'top' and 'bottom'. We could error out in the other cases.